### PR TITLE
[ADAM-1192] Correctly handle other whitespace in FASTA description.

### LINKED ISSUE #1192: MDTagging fails on GRCh38
I was running the `transform -add_md_tags` code on GRCh38, and was getting contig name mismatches that caused failures. It looked like a header parsing problem. TBD. I need to look back in the logs for the stack trace.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1192. More broadly handles whitespace in the FASTA description, and also disregards FASTA metadata which is inlined.

Testing this on the cluster as we speak.

### CODE REVIEW

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/Fasta2ADAMSuite.scala [hunk 1]
```diff
@@ -30,8 +30,8 @@ class Fasta2ADAMSuite extends ADAMFunSuite {
     val contigFragments = sc.loadParquetContigFragments(convertPath)
     assert(contigFragments.rdd.count() === 26)
     val first = contigFragments.rdd.first()
-    assert(first.getContig.getContigName === "gi|224384749|gb|CM000682.1|")
-    assert(first.getDescription === "Homo sapiens chromosome 20, GRCh37 primary reference assembly")
+    assert(first.getContig.getContigName === null)
```
Comment (heuermh) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/Fasta2ADAMSuite.scala]: Not sure I understand, the string before the first space is typically the sequence name in FASTA format.  You might want to compare to the various open-bio projects.

Comment (fnothaft) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/Fasta2ADAMSuite.scala]: Well, `gi` is the Entrez ID, and `gb` is the GenBank ID. I assure you that they are not the sequence name.

**Comment by fnothaft** [comment]
Whoops! Forgot to commit the test resource. Just pushed.

**Comment by fnothaft** [comment]
With this, recomputing the MD tags worked on the cluster with GRCh38.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**